src/components/select/select.component.ts

```diff
@@ -284,20 +286,25 @@ export class SelectComponent<T> implements OnInit, OnChanges, OnDestroy, Control
         // Standard keys for typeahead (up/down/esc)
         this._typeaheadKeyService.handleKey(event, this.singleTypeahead);
 
-        switch (event.keyCode) {
-            case ENTER:
-                if (this._dropdownOpen) {
-                    // Set the highlighted option as the value and close
-                    this.value = this.singleTypeahead.highlighted;
-                    this.dropdownOpen = false;
-                } else {
-                    this.dropdownOpen = true;
-                }
+        if (event.keyCode === ENTER) {
+            if (this._dropdownOpen) {
+                // Set the highlighted option as the value and close
+                this.value = this.singleTypeahead.highlighted;
+                this.dropdownOpen = false;
+            } else {
+                this.dropdownOpen = true;
+            }
+
+            // Update the input field. If dropdown isn't open then reset it to the previous value.
+            this.input = this.getDisplay(this.value);
+            event.preventDefault();
+        }
 
-                // Update the input field. If dropdown isn't open then reset it to the previous value.
-                this.input = this.getDisplay(this.value);
-                event.preventDefault();
-                break;
+        // when the user types and the value is not empty then we should open the dropdown
+        if (event.keyCode !== ESCAPE) {
+            // open the dropdown once the filter debounce has elapsed
+            this.filter$.pipe(take(1), takeUntil(this._onDestroy))
+                .subscribe(() => this.dropdownOpen = true);

```

Comment:
This doesn't look right to me, could be a memory leak. Instead of subscribing every time esc is pressed, can you not just subscribe once and set a flag here to check in the subscription?